modules/apps/object/object-rest-impl/src/main/java/com/liferay/object/rest/internal/vulcan/openapi/contributor/RelatedObjectEntryOpenAPIContributor.java

```diff
@@ -136,7 +136,7 @@ private void _contribute(
 				relatedObjectDefinition);
 
 		Map<String, Schema> relatedSchemas =
-			objectEntryOpenAPIResource.getSchemas();
+			objectEntryOpenAPIResource.getSchemas(relatedObjectDefinition);
```

Comment:
Careful here, if the objectEntryOpenAPIResource is not the one attached to the relatedObjectDefinition, we should get the proper objectEntryOpenAPIResource of the relatedObjectDefinition using the objectEntryOpenAPIResourceProvider